hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/common/LinkFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.restjava.common;
+
+import com.hedera.mirror.rest.model.Links;
+import java.util.function.Function;
+import org.springframework.data.domain.Pageable;
+
+public interface LinkFactory {
+    <T> Links create(Iterable<T> items, Pageable pageable, ParameterExtractor<T> extractor);
+
+    interface ParameterExtractor<T> {
+        Function<T, String> extract(String name);
```

Comment:
let's change this to

```suggestion
        String extract(T item, String name);
```

So you don't have to deal with the situation that there is a function to extract parameter `p`, however the extracted value is `null`.

With the new `extract` function, given the item, and parameter name `p`, if it returns `null`, just add all existing values for `p` to the next link.

Reply:
Got it I have made this change.